Updates to decentralizing-xmtp.md (#819)
xmtp/xmtp-dot-org: src/pages/decentralizing-xmtp.md

    - The `Payers` contract manages the payment of messaging fees to node operators, allowing applications and organizations to cover network costs without requiring users to pay directly.
    - The `IdentityUpdates` [contract](https://github.com/xmtp/xmtpd/blob/main/contracts/src/IdentityUpdates.sol) manages the list of wallet addresses associated with each XMTP inbox, enabling users to use multiple authorized wallet addresses to send and receive messages from a single inbox.
    - The `GroupMessages` [contract](https://github.com/xmtp/xmtpd/blob/main/contracts/src/GroupMessages.sol) manages group membership changes, ensuring that all clients have an identical view of group additions and removals in the correct order.

For more specifics, review [XIP-49](https://community.xmtp.org/t/xip-49-decentralized-backend-for-mls-messages/856) and follow the team's progress in the [xmtpd repo](https://github.com/xmtp/xmtpd/issues/118).

![Diagram of actors in the XMTP decentralized network](../../static/img/decentralizing-xmtp-1.png)

### Timeline and key milestones

> We launched a testnet in February, 2024. The testnet is currently unincentivized, with no fees charged to payers and no rewards distributed to node operators.

Reviewer: ```suggestion
We launched a testnet in February, 2025. The testnet is currently unincentivized, with no fees charged to payers and no rewards distributed to node operators.
```

XMTP's distributed architecture, use of smart contracts, and participation from multiple nodes and clients ensure that the network operates independently even if any single organization - including Ephemera - ceases to exist.

**17. What metadata is visible on the network?**

| Content | Is visible |
| --- | --- |
| When you joined the network | ✅ |
| When you add a new wallet or device to your account | ✅ |
| What time you have been invited to a conversation | ✅ |
| How many devices and wallets you have added to your account | ✅ |
| Who invited you to the conversation | ❌ |
| What conversations you have been invited to | ❌* |
| Who is in any group | ❌* |
| When you sent a message in a conversation | ❌ |
| The contents of the messages you send | ❌ |
| How many messages you have sent | ❌ |
| When you consent to join a conversation | ❌ |
| The name and description of any group | ❌ |
| The permissions and admin roles of any group | ❌ |

> **There is a risk that an attacker reading all messages on the network could draw statistical correlations between otherwise unrelated messages to reveal members that members were invited to a group at the same time, and which group ID they were added to, by looking at messages sent at similar times. We plan to address this risk by adding jitter to message timing before the release of XMTP mainnet.*

Reviewer: Is there an extra word here? "reveal members that members were invited"

Author: Good catch. Meaning is different depending on "that members" or "members that". I'll commit the former, since the latter implies member identities are revealed.

We believe the only true protection for your messages is strong encryption. Other messaging services may claim to have an additional layer of protection by limiting who can access the encrypted payloads and keeping them private, but this protection is prone to failure. Even the most trusted providers routinely get hacked and find the contents of their databases on the dark web. Government agencies find backdoors into supposedly closed networks. And when this happens, the hackers often find that there is lots of useful metadata - even in end-to-end encrypted systems - worth stealing. 

By keeping all the data in our network public our model is simple and transparent. Rather than hoping our systems will never get compromised, our goal is to make sure your private information stays private even if every node on our network has already been hacked. Anyone can inspect what data is available on the network, and security researchers can try their hardest to find vulnerabilities that compromise user security or privacy in the open. 

**19. Will XMTP support post quantum encryption?**

> Yes. Today there are no known quantum computers capable of decrypting messages encrypted with elliptic curves as large as the ones used in XMTP today, but we need to be prepared for the possibility that those computers are in development and will likely exist in the next decade. Given how cheap it is to store data, a well-funded adversary like a government may decide to keep a copy of some messages stored on our network today and wait for a computer capable of decrypting them. This is referred to as “Harvest Now Decrypt Later”. Other messaging services have recently started to roll out hybrid encryption schemes that include post quantum algorithms to protect against this threat.

Reviewer: ```suggestion
Yes. Today there are no known quantum computers capable of decrypting messages encrypted with elliptic curves as large as the ones used in XMTP today, but we need to be prepared for the possibility that those computers are in development and will likely exist in the next decade. Given how cheap it is to store data, a well-funded adversary like a government may decide to keep a copy of some messages stored on the network today and wait for a computer capable of decrypting them. This is referred to as “Harvest Now Decrypt Later”. Other messaging services have recently started to roll out hybrid encryption schemes that include post quantum algorithms to protect against this threat.
```

> We expect to complete the work needed to roll out Post Quantum encryption in 2025. MLS itself is very flexible with encryption schemes and makes it straightforward to migrate our ciphersuites to one that offers post quantum protection such as ML-KEM. The challenge lies in the payload sizes. Keys that are 32 bytes today can become multiple kilobytes. Ciphertext and signatures can double in size. 

Reviewer: ```suggestion
We expect to complete the work needed to roll out Post Quantum encryption in 2025. MLS itself is very flexible with encryption schemes and makes it straightforward to migrate the ciphersuites used by XMTP to one that offers post-quantum protection such as ML-KEM. The challenge lies in the payload sizes. Keys that are 32 bytes today can become multiple kilobytes. Ciphertext and signatures can double in size. 
```
